File: test/oob_capability_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:flutter_uwb/src/oob_capability.dart';

void main() {
  group('OobCapability', () {
    test('reserves stable wire values across the three peer kinds', () {
      expect(OobCapability.iosPeer, 0x01);
      expect(OobCapability.androidPeer, 0x02);
      expect(OobCapability.accessoryHost, 0x03);
    });

    test('falls back to Android peer when the byte is missing', () {
      expect(OobCapability.unknownDefault, OobCapability.androidPeer);
    });
  });
}

File: android/src/test/kotlin/com/ahmedhamdan/flutter_uwb/oob/OobCapabilityTest.kt
package com.ahmedhamdan.flutter_uwb.oob

import kotlin.test.Test
import kotlin.test.assertEquals

class OobCapabilityTest {

    @Test
    fun localServiceDataAdvertisesAndroidPeer() {
        val payload = OobCapability.localServiceData()
        assertEquals(1, payload.size)
        assertEquals(OobCapability.ANDROID_PEER, payload[0])
    }

    @Test
    fun peerBytesAreDistinct() {
        assertEquals(0x01.toByte(), OobCapability.IOS_PEER)
        assertEquals(0x02.toByte(), OobCapability.ANDROID_PEER)
    }
}
